File: CargoFlashCargoWebApps/JScript/Master/SPHCGroup.js

/*
*****************************************************************************
Javascript Name:	SPHCGroupJS     
Purpose:		    This JS used to get Grid data for SPHC Group Trans.
Company:		    CargoFlash Infotech Pvt Ltd.
Author:			    Tarun Kumar
Created On:		    21 May 2014
Updated By:         
Updated On:	        
Approved By:        
Approved On:	    
*****************************************************************************
*/
//var strData = [];
$(function () {
    debugger
    cfi.AutoCompleteV2("SHCCode", "SNo,Code", "SHCGroup_SHCCode", null, "contains", ",");
    var pageType = getQueryStringValue("FormAction").toUpperCase();
    if ((pageType != "READ" && pageType != "DELETE") && $("#Text_SHCCode").val() == "" && $("#SHCCode").val() != "") {
        var values = $('#SHCCode').val();
        $.ajax({
            url: "Services/Master/SPHCGroupService.svc/GetSPHCCodes",
            async: false,
            type: "POST",
            dataType: "json",
            cache: false,
            data: JSON.stringify({ Values: values }),
            contentType: "application/json; charset=utf-8",
            success: function (result) {
                debugger
                $("#Text_SHCCode").val(result);
            }
        });
    }
    cfi.BindMultiValue("SHCCode", $("#Text_SHCCode").val(), $("#SHCCode").val());
    $("#Name").on('drop', function () {
        return false;
    });
  
    $('#Name').on("contextmenu", function (e) {
        //alert('Right click disabled');
        return false;
    });

});
function ExtraCondition(textId) {
    var filterEmbargo = cfi.getFilter("AND");
    if (textId == "Text_SHCCode")
        return textId = cfi.getFilter("AND"), cfi.setFilter(textId, "SNo", "notin", $("#SHCCode").val()), cfi.autoCompleteFilter(textId);
}

//$.validator.addMethod(
//        "alphabetsOnly",
//        function (value, element, regexp) {
//            var re = new RegExp(regexp);
//            return this.optional(element) || re.test(value);
//        },
//        "Please check your input values again!!!."
//);
//$("#Name").rules("add", { alphabetsOnly: "^[a-zA-Z'.\\s]$" })

File: CargoFlash.ROMS.Model/Master/SPHCGroup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;

namespace CargoFlash.Cargo.Model.Master
{
    #region SPHC Group Description
    /*
	*****************************************************************************
	Class Name:		SPHCGroup   
	Purpose:		This class used to handle
	Company:		CargoFlash Infotech Pvt Ltd.
	Author:			Tarun Kumar
	Created On:		21 May 2014
    Updated By:    
	Updated On:
	Approved By:    
	Approved On:	
	*****************************************************************************
	*/
    #endregion

    [KnownType(typeof(SPHCGroup))]
    public class SPHCGroup
    {
        public int SNo { get; set; }
        public String Name { get; set; }
        public string SHCCode { get; set; }
        public string Text_SHCCode { get; set; }
        public bool IsActive { get; set; }
        public string CreatedBy { get; set; }
        public string UpdatedBy { get; set; }
        public String Active { get; set; }
        
    }

    //[KnownType(typeof(SPHCGroupTrans))]
    //public class SPHCGroupTrans
    //{
    //    //public int SNo { get; set; }
    //    public int SPHCGroupSNo { get; set; }
    //    public String SPHCSNo { get; set; }
    //    //public int IsActive { get; set; }
    //    //public string CreatedBy { get; set; }
    //    //public string UpdatedBy { get; set; }
    //    //public String Active { get; set; }
    //    public String HdnSPHCSNo { get; set; }
    //}
}
